### ISSUE TITLE
Optimize controller UI for large cluster/tables

### ISSUE THREAD

**Issue by dachaostripe** [apache/pinot#10504]
We have large tables that have over 220k and 370k offline segments, that we almost always timeout trying to open the offline table page, as it seems the page was trying to load all segment data on the page. It would be great if the UI can be redesigned or tweaked to not trying to load all those data on page load, to make UI more responsive.

**Comment by Jackie-Jiang** [comment]
cc @joshigaurava @jayeshchoudhary @jadami10

**Comment by jadami10** [comment]
yup, i think we can take the same strategy as the tables page and render everything separately

**Comment by joshigaurava** [comment]
@Jackie-Jiang, noted.

**Comment by jayeshchoudhary** [comment]
I think this is more related to the backend. Currently, UI calls idealState and externalView APIs to show segments. 
If there are 100 segments the API response will be faster, if there are 100k segments the API response will be slow. so UI can't do anything here to compensate for the slowness.

IMO we should have some pagination/search in the API, which UI can leverage to only show a subset of the segments. 
But, I don't know the userflow here. maybe we always want to look at all the segments? or if we show only a subset of segments how do we consume them?

We can think about the common workflows here and decide based on user behavior. After this, there will be backend + UI changes.

**Comment by Jackie-Jiang** [comment]
@jayeshchoudhary Can you help list down all the data that the UI is collecting before the page is rendered? Then we can discuss how to reduce the data to be fetched for large tables.

**Comment by jadami10** [comment]
I'm hoping to take a pass at this during the EOY slowdown. My approach is going to be:
- decouple rendering from data loading
- use loading skeletons while loading data

I'm going to tackle the biggest known issues first:
- /size: this reaches out to servers for table size. and depending on current load, it can take ~forever. It also only populates numbers in the UI, so there's no reason to stop rendering the whole page
- idealstate/externalview: this is known to be slow for large tables. It also causes memory issues in the browser tab, but i don't have an easy solution for that yet until things like the segment list do server side pagination. But this endpoint is mainly used to show the "status" of the table or the segments. Again, no need to block the whole UI from loading.
